File: ruby/spec/parser/continuations.spec.rb
# frozen_string_literal: true

input = <<~DOC.strip
field: value
\\ spaced continuation
| direct continuation

field_with_attribute:
attribute = value
\\ spaced continuation
| direct continuation

field_with_item:
- value
\\ spaced continuation
| direct continuation

field:
\\ spaced continuation
| direct continuation

field_with_attribute:
attribute =
\\ spaced continuation
| direct continuation

field_with_item:
-
\\ spaced continuation
| direct continuation
DOC

describe Enolib::Parser do
  describe 'Continuations' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end

File: javascript/specs/unpack/continuations.spec.js
import { parse } from '../..';
import { unpack } from '../unpack.js';

const input = `
field_with_value: [value]
\\ [spaced_continuation]
| [direct_continuation]

field_with_attribute:
attribute = [value]
\\ [spaced_continuation]
| [direct_continuation]

field_with_item:
- [item]
\\ [spaced_continuation]
| [direct_continuation]

empty_field:
\\ [spaced_continuation]
| [direct_continuation]

field_with_empty_attribute:
empty_attribute =
\\ [spaced_continuation]
| [direct_continuation]

field_with_empty_item:
-
\\ [spaced_continuation]
| [direct_continuation]
`.trim();

describe('Blackbox test', () => {
    describe('Continuations', () => {
        it('performs as expected', () => {
            const document = parse(input);
            expect(unpack(document)).toMatchSnapshot();
        });
    });
});
